### PR TITLE
PR Issue 2431

### LINKED ISSUE #2431: Shadow variable
/SofaKernel/modules/SofaHelper/src/sofa/helper/system/PipeProcess.cpp line 202
Code:
  for(int i=0;; ++i)
        {
            GetExitCodeProcess(piProcInfo.hProcess,&exit);      //while the process is running
            if (exit != STILL_ACTIVE)
                break;

            bool busy = false;
            for (int i=0; i<2; ++i) // <- shadow variable
            {

### PR DESCRIPTION
**Pull request by maddin200**
fix shadow variable

Fix #2431 

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][force-full-build]

File: SofaKernel/modules/SofaHelper/src/sofa/helper/system/PipeProcess.cpp [hunk 1]
```diff
@@ -192,7 +192,8 @@ bool PipeProcess::executeProcess(const std::string &command,  const std::vector<
         unsigned long exit = 0;
         for (int i=0; i<2; ++i)
             nfill[i] = 0;
-        for(int i=0;; ++i)
+        
+        for(int jj=0;; ++jj)
```
Comment (alxbilger) [inline SofaKernel/modules/SofaHelper/src/sofa/helper/system/PipeProcess.cpp:196]: I don't know this algorithm, but from what I see, `jj` is not used and there is not end condition. So I would replace it by `while(true)`

**Comment by maddin200** [comment]
Should be the standard solution for those kind of loops in the project.

**Comment by maddin200** [comment]
Leave both PRs unchanged and later on (in a new PR) replace all patterns like for(; ;) with while(true).

**Comment by fredroy** [comment]
Fixed later in https://github.com/sofa-framework/sofa/pull/2477 , closing

**Pull request closed without merging**